### PR TITLE
Added CI and Code Coverage toolings

### PR DESCRIPTION
**Pull request by mojoX911**
This adds CI (build + test) using github action.

The tests in nightly build produces coverage report which can be displayed using codecov

Action for maintainer: 

- Sign up in codecov (https://about.codecov.io/)
- Install codecov extension in github. Add the repo reading permission (https://github.com/marketplace/codecov)
- Setup the codecov environment secret for actions to upload coverage report.

Rest is taken care of by the scripts.

This will create CI testing report as well as code coverage report for each PR.

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -0,0 +1,1267 @@
+# This file is automatically @generated by Cargo.
```
Comment (chris-belcher) [inline Cargo.lock:1]: Looks like you accidentality uploaded this file as well?

**Comment by chris-belcher** [comment]
Thanks for this. I'll set it up soon.

File: Cargo.lock (hunk 1 above)
Comment (mojoX911) [reply Cargo.lock]: Ya this was unintentional. Sorry. Reverted it back in latest push.

**Comment by mojoX911** [comment]
Refactored the action files. Now there are two files, one builds and tests with stable + nightly. Another runs the tests and generate and upload code coverage report.

**Comment by chris-belcher** [comment]
Sorry for not getting back to this sooner. I don't know enough about CI to review this. If you're confident it's fine would you like to set it up? I can add you as another maintainer.

**Comment by mojoX911** [comment]
No Issues. I wont mind setting it up myself. I think it will help a lot in future merging process.

**Pull request merged**